pubtools/pulplib/_impl/client/client.py

```diff
@@ -179,6 +182,28 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def _do_upload_file(self, upload_id, file_obj):
+        def do_next_upload(checksum, size):
+            data = file_obj.read(self._CHUNK_SIZE)
+            if data:
+                checksum.update(data)
+                return f_flat_map(
+                    self._do_upload(data, upload_id, size),
+                    lambda _: do_next_upload(checksum, size + len(data)),
+                )
+            else:
+                # nothing more to upload, return checksum and size
+                if is_string:
+                    file_obj.close()
+                return f_return((checksum.hexdigest(), size))
+
+        is_string = isinstance(file_obj, six.string_types)
```

Comment:
I think this is still not right.  I pointed to six.string_types doc but didn't expect you can just plug it in and it'll do what's needed, because it still doesn't work for bytes on Python3 (right?)

I thought you'll probably need to check both string_types and binary_type, if checking this way.

Alternatively, you can invert this check and decide that file_obj is a file object if and only if `'close' in dir(file_obj)` or similar.

Reply:
Yeah, this confused me a bit. 
I thought bytes just wouldn't be used as a string anymore in python3.